widget/ui/src/components/Toast/Toast.Provider.tsx

```diff
@@ -0,0 +1,84 @@
+import type {
+  ProviderContext,
+  ProviderPropTypes,
+  ToastProps,
+  ToastType,
+} from './Toast.types';
+import type { PropsWithChildren } from 'react';
+
+import React, { createContext, useContext, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Toast } from './Toast';
+import { Wrapper } from './Toast.styles';
+
+const ToastContext = createContext<ProviderContext | undefined>(undefined);
+
+let id = 1;
+
+export const ToastProvider = (props: PropsWithChildren<ProviderPropTypes>) => {
+  const {
+    children,
+    anchorOrigin = {
+      horizontal: 'right',
+      vertical: 'bottom',
+    },
+    container,
+    style = {},
+  } = props;
+  const [toasts, setToasts] = useState<ToastProps[]>([]);
+
+  // eslint-disable-next-line react/jsx-no-constructed-context-values
+  const api: ProviderContext = {
+    addToast(content: ToastType) {
+      setToasts((toasts) => [
```

Comment:
Prevent user to send non-unique ids here so you can remove line `48` to `55`.
Please add a warning to let the user know we are omitting the request.
```
if( alreadySent ){
console.warn("Has been sent before.")
return;
}
```

Reply:
fixed.

Reply:
Sorry it seems I made you confused a bit. My text and naming wasn't accurate and just wanted to show the idea. you can use `isToastExistById` and for warning message you use: `You are trying to send a toast with an id that already exists. Please change your toast id or you can use it when toast has been hide. `

Reply:
changed.